src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -284,6 +290,59 @@ protected override void Initialize()
         }
 
         /// <summary>
+        /// Re-applies in-memory profiles to the newly created snapshot
+        /// </summary>
+        protected void MergeExistingInMemoryProfiles(LaunchSettingsData newSnapshot, ILaunchSettings prevSnapshot)
+        {
+            for (int i =0; i < prevSnapshot.Profiles.Count; i++)
+            {
+                var profile = prevSnapshot.Profiles[i];
+                if(profile.IsInMemoryObject())
+                {
+                    // Does it already have one with this name?
+                    if(newSnapshot.Profiles.FirstOrDefault(p => LaunchProfile.IsSameProfileName(p.Name, profile.Name)) == null)
+                    {
+                        // Create a new one from the existing in-memory profile and insert it in the same location, or the end if it
+                        // is beyond the end of the list
+                        if(i > newSnapshot.Profiles.Count)
+                        {
+                            newSnapshot.Profiles.Add(LaunchProfileData.FromILaunchProfile(profile));
+                        }
+                        else
+                        {
+                            newSnapshot.Profiles.Insert(i, LaunchProfileData.FromILaunchProfile(profile));
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Re-applies in-memory global options to the newly created snapshot
+        /// </summary>
+        protected void MergeExistingInMemoryGlobalSettings(LaunchSettingsData newSnapshot, ILaunchSettings prevSnapshot)
+        {
+            if(prevSnapshot.GlobalSettings != null)
+            {
+                foreach (var kvp in prevSnapshot.GlobalSettings)
+                {
+                    if(kvp.Value.IsInMemoryObject())
+                    {
+                        if(newSnapshot.OtherSettings == null)
+                        {
+                            newSnapshot.OtherSettings = new Dictionary<string, object> ();
+                            newSnapshot.OtherSettings[kvp.Key] = kvp.Value;
+                        }
+                        else if(!newSnapshot.OtherSettings.TryGetValue(kvp.Key, out var existingValue))
```

Comment:
>!newSnapshot.OtherSettings.TryGetValue(kvp.Key [](start = 32, length = 46)

if key exist we would not update it's value?

Reply:
Correct. Disk values trump in-memory settings